hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/upgrade/OmVersionFactory.java

```diff
@@ -0,0 +1,133 @@
+    registerClientRequest(Type.DeleteVolume.name(), OMVolumeDeleteRequest::new);
+    registerClientRequest(Type.CreateBucket.name(), OMBucketCreateRequest::new);
+    registerClientRequest(Type.DeleteBucket.name(), OMBucketDeleteRequest::new);
+    registerClientRequest(Type.SetBucketProperty.name(),
+        OMBucketSetPropertyRequest::new);
+    registerClientRequest(Type.AllocateBlock.name(),
+        OMAllocateBlockRequest::new);
+    registerClientRequest(Type.CreateKey.name(), OMKeyCreateRequest::new);
+    registerClientRequest(Type.CommitKey.name(), OMKeyCommitRequest::new);
+    registerClientRequest(Type.DeleteKey.name(), OMKeyDeleteRequest::new);
+    registerClientRequest(Type.DeleteKeys.name(), OMKeysDeleteRequest::new);
+    registerClientRequest(Type.RenameKey.name(), OMKeyRenameRequest::new);
+    registerClientRequest(Type.CreateDirectory.name(),
+        OMDirectoryCreateRequest::new);
+    registerClientRequest(Type.CreateFile.name(), OMFileCreateRequest::new);
+    registerClientRequest(Type.PurgeKeys.name(), OMKeyPurgeRequest::new);
+    registerClientRequest(Type.InitiateMultiPartUpload.name(),
+        S3InitiateMultipartUploadRequest::new);
+    registerClientRequest(Type.CommitMultiPartUpload.name(),
+        S3MultipartUploadCommitPartRequest::new);
+    registerClientRequest(Type.AbortMultiPartUpload.name(),
+        S3MultipartUploadAbortRequest::new);
+    registerClientRequest(Type.CompleteMultiPartUpload.name(),
+        S3MultipartUploadCompleteRequest::new);
+    registerClientRequest(Type.AddAcl.name(),
+        OzoneManagerRatisUtils::getOMAclRequest);
+    registerClientRequest(Type.RemoveAcl.name(),
+        OzoneManagerRatisUtils::getOMAclRequest);
+    registerClientRequest(Type.SetAcl.name(),
+        OzoneManagerRatisUtils::getOMAclRequest);
+    registerClientRequest(Type.GetDelegationToken.name(),
+        OMGetDelegationTokenRequest::new);
+    registerClientRequest(Type.CancelDelegationToken.name(),
+        OMCancelDelegationTokenRequest::new);
+    registerClientRequest(Type.RenewDelegationToken.name(),
+        OMRenewDelegationTokenRequest::new);
+    registerClientRequest(Type.GetS3Secret.name(), S3GetSecretRequest::new);
+    registerClientRequest(Type.RecoverTrash.name(), OMTrashRecoverRequest::new);
+
+    // Adding "Mock" Create Key Request V2.
+    registerClientRequest(Type.CreateKey.name(), OMLayoutFeature.NEW_FEATURE,
+        OMKeyCreateRequestV2::new);
+  }
+
+  public void registerClientRequest(
+      String type, OMLayoutFeature layoutFeature,
+      Function<OMRequest, OMClientRequest> function) {
+    VersionFactoryKey key = new VersionFactoryKey.Builder()
+        .key(type).version(layoutFeature.layoutVersion()).build();
+    writeRequestFactory.register(key, function);
+  }
+
+  public OMClientRequest getClientRequest(OMRequest omRequest) {
+    Function<OMRequest, ? extends OMClientRequest> instance =
+        writeRequestFactory.getInstance(getVersionFactoryKey(omRequest));
```

Comment:
When input omRequest type is not existed, we should return IllegalStateException error as original logic OzoneManagerRatisUtils#createClientRequest did.